Pod/Classes/UISideMenuNavigationController.swift

```diff
@@ -147,17 +147,52 @@ open class UISideMenuNavigationController: UINavigationController {
         if SideMenuManager.menuAllowPopIfPossible {
             for subViewController in navigationController.viewControllers {
                 if type(of: subViewController) == type(of: viewController) {
+                    if SideMenuManager.menuTabMode {
+                        navigationController.popToViewController(subViewController, animated: false)
+                        CATransaction.commit()
+                        return
+                    }
                     navigationController.popToViewController(subViewController, animated: animated)
                     CATransaction.commit()
                     return
                 }
             }
         }
         
+        if SideMenuManager.menuPreserveViewOnPush {
+            var viewControllers = navigationController.viewControllers
+            let filtered = viewControllers.filter{$0.restorationIdentifier == viewController.restorationIdentifier}
+            if let preservedView = filtered.first {
+                viewControllers = viewControllers.filter() { $0 !== preservedView }
+                viewControllers.append(preservedView)
+                if SideMenuManager.menuTabMode {
+                    navigationController.setViewControllers(viewControllers, animated: false)
```

Comment:
You're welcome to take a stab at it! Truthfully I haven't fully thought through how that should be architected or the best names yet so I might need to modify what you put together to something I think makes more sense on review.

At a high level, there should be some single properly, like `menuPushStyle` or `menuPushBehavior`, that lets the developer specify how push calls are handled. Those values/behaviors might be (names subject to change):
* **.default** // Default behavior
* **.popToSameClassWhenPossible** // The behavior of `menuAllowPopIfPossible`
* **.preventDoublePush** // The behavior of `menuAllowPushOfSameClassTwice`
* **.replace** // Releases the current view controller and pushes the new view controller
* **.preserve** // The current behavior for `menuPreserveOnPush` without setting `hidesBackButton = true`
* **.preserveAndHideBackButton** // The current behavior `menuPreserveOnPush`

Since this would be making some properties obsolete, there would need to be some deprecation notices put in for the replaced properties (see code for deprecated `menuAnimationTransformScaleFactor`) while still making them work as expected by appropriately setting the new property to the desired value.